### ISSUE TITLE
Multi-process application dependency problem

### ISSUE THREAD

**Issue by igor523** [dotnet/cli#7234]
Hello!
I'm trying to develop multi-process application - some parent process starts child processes to do the job. I'm using Visual Studio to create and debug project, but this behavior preserves when launching parent process through dotnet run.

## Steps to reproduce
1. Create a solution and 2 console application in it. Lets assume that one of them is Parent and other is Child.
2. Add some nuget dependency to Child project. For this description, I'll use Microsoft.Extensions.Logging v1.1.2, but it doesn't really matter.
3. Add Child to Parent's dependencies, so build process could copy Child's files to Parent's build path.
4. Write some code in Child using nuget package installed.
5. Write code in Parent to launch Child using Process.Start
6. Build and run Parent.

## Expected  behavior
Parent runs and launches Child process, which is should be running fine.

## Actual behavior
Child process throws exception: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Logging.Abstractions

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.4)

Product Information:
 Version:            1.0.4
 Commit SHA-1 hash:  af1e6684fd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16237
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.4

Code used in Parent:
```c#
using System.Diagnostics;
using System.IO;
using System.Reflection;
using System.Threading;

namespace MultiProcessParent
{
    class Program
    {
        static void Main(string[] args)
        {
            //Little hack to simplify Start invokation
            Directory.SetCurrentDirectory(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location));
            Process.Start("MultiProcessChild.dll");

            Thread.CurrentThread.Join();
        }
    }
}
```

Code used in Child
```c#
using System;
using System.Threading;
using Microsoft.Extensions.Logging;

namespace MultiProcessChild
{
    class Program
    {
        static void Main(string[] args)
        {
            var loggerFactory = new LoggerFactory().AddConsole();
            var logger = loggerFactory.CreateLogger("Test");
            logger.LogError("Error!");

            Thread.CurrentThread.Join();
        }
    }
}
```

Maybe it's my mistake? What am I doing wrong? Is there some other, more "right" way, to do this?
My spelling or grammar could be wrong somewhere, sorry for that

Thanks!

**Comment by livarcocc** [comment]
The problem here is that when you reference the dependency, it gets copied over to the application as a library (without all the files it needs to run).

I would even be careful of building both apps to the same folder, because if the apps have similar dependencies but with different versions, they may stomp on each other.

You can build boths apps, putting them in a known location and then you can run them. However, when getting ready for production, what you really need to do is publish them both to separate folders.

**Comment by igor523** [comment]
Dependency versions is good point, didn't thought about that earlier.
But the question is there - how can I debug this kind of apps? In test scenario both projects doesn't have common dependencies, so this is definitely not version mismatch. 
By the way, i can't find any of dependency dll in build folders that Visual Studio (and dotnet build command) uses. In Child build directory there is 1 dll, representing Child app itself, and no other (Microsoft.Extensions.Logging in this case) dll is there, but somehow Studio or dotnet run handles this in one-process case, but not in multi-process, and that thing is little confusing... 
So, in other words, when I add dependencies to Parent and launches it via Studio or dotnet run everything is fine and dependencies are loaded well. But if Parent starts Child process, its deps are gone somewhere. Am I missing something?

**Comment by igor523** [comment]
In case I didn't said that clear or that fact drowned in my poor English - I'm talking about debugging and development, not publishing. I know that publishing copies all dependencies to a single dir, and this solves problem. But publishing should be used when app is ready to launch in production, right?

**Comment by livarcocc** [comment]
The problem here is that we don't have a mechanism in place today where you reference a project targeting netcoreapp and outputtype exe and we understand that to mean that you want the projectreference to be copied to the build output as an application itself. We treat project references as libraries, and as such, during build, you end up missing necessary files by the application.

What I would suggest, even in this case, is that you publish the second application in a different location and have the first one invoke it, this way, you can debug the first app (if that's what you are trying to do) even when it is invoking the second one..

**Comment by livarcocc** [comment]
Closing this issue as we have no intention at the moment to support build a project reference to another application as an application inside another.

Please, re-activate if the the suggestions I made above do not work for you.
